src/meta/ActiveHostsMan.cpp

```diff
@@ -17,15 +18,42 @@ namespace meta {
 kvstore::ResultCode ActiveHostsMan::updateHostInfo(kvstore::KVStore* kv,
                                                    const HostAddr& hostAddr,
                                                    const HostInfo& info,
-                                                   const LeaderParts* leaderParts) {
+                                                   const AllLeaders* allLeaders) {
     CHECK_NOTNULL(kv);
     std::vector<kvstore::KV> data;
     data.emplace_back(MetaServiceUtils::hostKey(hostAddr.host, hostAddr.port),
                       HostInfo::encodeV2(info));
-    if (leaderParts != nullptr) {
-        data.emplace_back(MetaServiceUtils::leaderKey(hostAddr.host, hostAddr.port),
-                          MetaServiceUtils::leaderVal(*leaderParts));
+    std::vector<std::string> leaderKeys;
+    std::vector<int64_t> terms;
+    if (allLeaders != nullptr) {
+        for (auto& spaceLeaders : *allLeaders) {
+            auto spaceId = spaceLeaders.first;
+            for (auto& partLeader : spaceLeaders.second) {
+                auto key = MetaServiceUtils::leaderKey(spaceId, partLeader.get_part_id());
+                leaderKeys.emplace_back(std::move(key));
+                terms.emplace_back(partLeader.get_term());
+            }
+        }
+        auto keys = leaderKeys;
+        std::vector<std::string> values;
+        auto ret = kv->multiGet(kDefaultSpaceId, kDefaultPartId, std::move(keys), &values);
```

Comment:
Here you need to judge whether `ret.first == kvstore::ResultCode::SUCCEEDED`
If they are equal, do the following

Reply:
good catch, fixed